### PR TITLE
gh-101282: Enclose BOLT_APPLY_FLAGS value in double quotes

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-101282

Without this revert on macOS:
````
checking for --enable-bolt... no
checking BOLT_INSTRUMENT_FLAGS...
checking BOLT_APPLY_FLAGS... ./configure: line 8401: -reorder-blocks=ext-tsp: command not found

````

With this revert on macOS:
````
checking BOLT_INSTRUMENT_FLAGS...
checking BOLT_APPLY_FLAGS... -update-debug-sections -reorder-blocks=ext-tsp -reorder-functions=hfsort+ -split-functions -icf=1 -inline-all -split-eh -reorder-functions-use-hot-size -peepholes=none -jump-tables=aggressive -inline-ap -indirect-call-promotion=all -dyno-stats -use-gnu-stack -frame-opt=hot

### CODE REVIEW

**Comment by corona10** [comment]
cc @indygreg  Not decided yet which PR will be merged

**Comment by erlend-aasland** [comment]
Dong-hee: I'll take a look tonight (CET).

File: configure.ac [hunk 1]
```diff
@@ -2087,7 +2087,7 @@ if test -z "${BOLT_APPLY_FLAGS}"
 then
   AS_VAR_SET(
     [BOLT_APPLY_FLAGS],
-    [m4_join([ ],
+    ["m4_join([ ],
```
Comment (erlend-aasland) [inline configure.ac]: This looks awkward to me; it is better Autoconf practice to use:

```m4
    [m4_normalize("
      -update-debug-sections
      -reorder-blocks=ext-tsp
      ...
    ")]

**Review by erlend-aasland — CHANGES_REQUESTED** [review]
https://github.com/python/cpython/pull/104752/files#r1203531051

File: configure.ac (hunk 1 above)
Comment (erlend-aasland) [reply configure.ac]: If this suggestion is applied, I prefer this option to #104751

Comment (corona10) [reply configure.ac]: > This looks awkward to me; it is better Autoconf practice to use:

Yeah agree.

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks!

**Comment by erlend-aasland** [comment]
I forgot to ask the obvious question: did you verify that the warning disappeared?

**Pull request merged**